src/PhpSpec/Matcher/ObjectStateMatcher.php

```diff
@@ -120,15 +120,15 @@ public function getPriority(): int
      * @param boolean  $expectedBool
      * @param mixed    $result
      *
-     * @return FailureException
+     * @return NotEqualException
      */
-    private function getFailureExceptionFor(callable $callable, bool $expectedBool, $result): FailureException
+    private function getNotEqualExceptionFor(callable $callable, bool $expectedBool, $result): NotEqualException
     {
-        return new FailureException(sprintf(
+        return new NotEqualException(sprintf(
```

Comment:
I feel like this is overloading NotEqualException TBH, it might be better to have a new exception type that's handled in MethodReturnedNullListener?

Reply:
I have added a MethodFailureException class in a82ffef0842e1ba0744276dd7fbed6bc03105a2f - hope this is what you were meaning?